packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -495,7 +495,7 @@ export function useConnections({
         id: new UUID().toString(),
       };
 
-      if (!duplicate.favorite) {
+      if (!duplicate.favorite || !duplicate.favorite.name) {
         duplicate.favorite = { name: getConnectionTitle(duplicate) };
```

Comment:
Probably should merge so that we don't loose the color in case it was selected, but the name wasn't?

```suggestion
        duplicate.favorite = { ...duplicate.favorite, name: getConnectionTitle(duplicate) };
```